src/Artemis.Core/Services/WebServer/WebServerService.cs

```diff
@@ -83,76 +94,64 @@ private void PluginManagementServiceOnPluginFeatureDisabled(object? sender, Plug
             _controllers.RemoveAll(c => c.Feature == e.PluginFeature);
         }
 
-        if (_modules.Any(m => m.Feature == e.PluginFeature))
-        {
-            mustRestart = true;
-            _modules.RemoveAll(m => m.Feature == e.PluginFeature);
-        }
-
         if (mustRestart)
-            StartWebServer();
+            _ = StartWebServer();
     }
 
     /// <inheritdoc />
     public void Dispose()
     {
-        Server?.Dispose();
+        Server?.DisposeAsync();
         _webServerPortSetting.SettingChanged -= WebServerPortSettingOnSettingChanged;
     }
 
-    public WebServer? Server { get; private set; }
-    public PluginsModule PluginsModule { get; }
+    public IServer? Server { get; private set; }
+    public PluginsHandler PluginsHandler { get; }
     public event EventHandler? WebServerStarting;
 
 
     #region Web server managament
 
-    private WebServer CreateWebServer()
+    private async Task<IServer> CreateWebServer()
     {
         if (Server != null)
         {
-            if (_cts != null)
-            {
-                _cts.Cancel();
-                _cts = null;
-            }
-
-            Server.Dispose();
+            await Server.DisposeAsync();
             OnWebServerStopped();
             Server = null;
         }
 
-        WebApiModule apiModule = new("/", SystemTextJsonSerializer);
-        PluginsModule.ServerUrl = $"http://localhost:{_webServerPortSetting.Value}/";
-        WebServer server = new WebServer(o => o.WithUrlPrefix($"http://*:{_webServerPortSetting.Value}/").WithMode(HttpListenerMode.EmbedIO))
-            .WithLocalSessionManager()
-            .WithModule(PluginsModule);
-
-        // Add registered modules
-        foreach (WebModuleRegistration? webModule in _modules)
-            server = server.WithModule(webModule.CreateInstance());
+        PluginsHandler.ServerUrl = $"http://localhost:{_webServerPortSetting.Value}/";
+        
+        LayoutBuilder serverLayout = Layout.Create()
+            .Add(PluginsHandler)
+            .Add(ErrorHandler.Structured())
```

Comment:
The structured error handler is the default one so this call should not be needed here.